sdcm/nemesis.py

```diff
@@ -2398,6 +2399,163 @@ def disrupt_memory_stress(self):
         self.target_node.remoter.run(
             "stress-ng --vm-bytes $(awk '/MemTotal/{printf \"%d\\n\", $2 * 0.9;}' < /proc/meminfo)k --vm-keep -m 1 -t 100")
 
+    def disrupt_toggle_cdc_feature_properties_on_table(self):
+        """Manupulate cdc feature settings
+
+        randomly choose table form list and toggle cdc feature settings
+        if cdc was disaled, wait settings applying and scylla_cdc_log table
+        removing and enable cdc back
+        if cdc was enabled, run cdc stressor tool for table for period [5-20] minutes
+
+        """
+        self._set_current_disruption("ToggleCDCProperties")
+
+        def _generate_cdc_properties(cdc_settings: dict) -> Dict[str, str]:
+            cdc_settings['enabled'] = False if cdc_settings['enabled'] else True
+
+            if cdc_settings['enabled']:
+                cdc_settings['preimage'] = random.choice([True, False])
+                cdc_settings['postimage'] = random.choice([True, False])
+                cdc_settings['ttl'] = str(random.randint(300, 3600))
+
+            return cdc_settings
+
+        ks_tables_with_cdc = self.cluster.get_all_tables_with_cdc(self.target_node)
+        self.log.debug(f"Found next tables with enabled cdc feature: {ks_tables_with_cdc}")
+
+        if ks_tables_with_cdc:
+            ks, table = random.choice(ks_tables_with_cdc).split(".")
+        else:
+            ks_tables = self.cluster.get_non_system_ks_cf_list(self.target_node, filter_out_mv=True)
+            ks, table = random.choice(ks_tables).split(".")
+
+        initial_cdc_settings = self.cluster.get_table_cdc_properties(self.target_node, ks, table)
+        self.log.debug(f"Initial cdc settings: {initial_cdc_settings}")
+        new_cdc_settings = _generate_cdc_properties(initial_cdc_settings)
+        self.log.debug(f"New cdc settings: {new_cdc_settings}")
+
+        self.log.info(f"Apply new cdc settigs for table {ks}.{table}: {new_cdc_settings}")
+        self._alter_table_with_cdc_properties(ks, table, new_cdc_settings)
+        self._verify_cdc_feature_status(ks, table, new_cdc_settings)
+
+        if new_cdc_settings["enabled"]:
+            self.log.info("Run cdc_stressor tool")
+            self._run_cdc_stressor_tool(ks, table)
+        else:
+            self.log.info("Let work cluster with new cdc settigns")
+            time.sleep(300)
+
+        self.log.info(f'Restore cdc settings for {ks}.{table} with initial state: {initial_cdc_settings}')
+        self._alter_table_with_cdc_properties(ks, table, initial_cdc_settings)
+        self._verify_cdc_feature_status(ks, table, initial_cdc_settings)
```

Comment:
I think to make the nemesis simpler and debuggable, we can give up on disabling and enabling the CDC itself, and just toggle the properties (even only one of them each time).

Reply:
Done